Reduce InjectedBundlePageUIClient API surface area

achristensen07 opened the pull request:
#### 510548af7b7d87d3b26461a9c04a3ca6aced9f63
<pre>
Reduce InjectedBundlePageUIClient API surface area
<a href="https://rdar.apple.com/159335767">rdar://159335767</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=298012">https://bugs.webkit.org/show_bug.cgi?id=298012</a>

Reviewed by Jeff Miller.

With <a href="https://rdar.apple.com/143399667">rdar://143399667</a> and <a href="https://rdar.apple.com/159318803">rdar://159318803</a> fixed, most of the callbacks are unused or unreachable.
This removes the unused code, removes tests that were just testing it, and adopts alternatives
for tests that were using it to test other things.

* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::addMessageWithArgumentsToConsole): Deleted.
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration _setShouldSendConsoleLogsToUIProcessForTesting:]):
(-[WKWebViewConfiguration _shouldSendConsoleLogsToUIProcessForTesting]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::addMessageToConsoleForTesting):
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::mouseDidMoveOverElement):
(API::InjectedBundle::PageUIClient::didClickAutoFillButton):
(API::InjectedBundle::PageUIClient::willAddMessageWithArgumentsToConsole): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptAlert): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptConfirm): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptPrompt): Deleted.
(API::InjectedBundle::PageUIClient::pageDidScroll): Deleted.
(API::InjectedBundle::PageUIClient::didExceedDatabaseQuota): Deleted.
(API::InjectedBundle::PageUIClient::plugInStartLabelTitle const): Deleted.
(API::InjectedBundle::PageUIClient::plugInStartLabelSubtitle const): Deleted.
(API::InjectedBundle::PageUIClient::plugInExtraStyleSheet const): Deleted.
(API::InjectedBundle::PageUIClient::plugInExtraScript const): Deleted.
(API::InjectedBundle::PageUIClient::didResignInputElementStrongPasswordAppearance): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.cpp:
(WebKit::InjectedBundlePageUIClient::willAddMessageToConsole): Deleted.
(WebKit::InjectedBundlePageUIClient::willAddMessageWithArgumentsToConsole): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptAlert): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptConfirm): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptPrompt): Deleted.
(WebKit::InjectedBundlePageUIClient::pageDidScroll): Deleted.
(WebKit::InjectedBundlePageUIClient::didExceedDatabaseQuota): Deleted.
(WebKit::InjectedBundlePageUIClient::didResignInputElementStrongPasswordAppearance): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::addMessageToConsole):
(WebKit::WebChromeClient::runJavaScriptAlert):
(WebKit::WebChromeClient::runJavaScriptConfirm):
(WebKit::WebChromeClient::runJavaScriptPrompt):
(WebKit::WebChromeClient::inputElementDidResignStrongPasswordAppearance):
(WebKit::WebChromeClient::addMessageWithArgumentsToConsole): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::pageDidScroll):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/BundlePageConsoleMessage.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/BundlePageConsoleMessageWithDetails.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ConsoleMessageWithDetails.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DidResignInputElementStrongPasswordAppearance.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::webViewWithOpenInspector):
(TestWebKitAPI::TEST(PrivateClickMeasurement, DaemonDebugMode)):
(TestWebKitAPI::setupSKAdNetworkTest):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetwork)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetworkAboutBlank)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetworkWithoutNavigatingToAppStoreLink)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, NetworkProcessDebugMode)):
(TestWebKitAPI::setInjectedBundleClient): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FTP.mm:
(TestWebKitAPI::TEST(WKWebView, FTPMainResource)):
(TestWebKitAPI::TEST(WKWebView, FTPMainResourceRedirect)):
(TestWebKitAPI::(, )):
(TestWebKitAPI::FTPSubresourceRedirect)):
(TestWebKitAPI::didReceivePageMessageFromInjectedBundle): Deleted.
(TestWebKitAPI::setInjectedBundleClient): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/LegacyPDFPluginTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFTestHelpers.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenTypeDidChange)): Deleted.
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenValueDidChange)): Deleted.
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenFormIsReset)): Deleted.
* Tools/TestWebKitAPI/cocoa/TestUIDelegate.h:
* Tools/TestWebKitAPI/cocoa/TestUIDelegate.mm:
(-[TestUIDelegate _webView:didReceiveConsoleLogForTesting:]):

Canonical link: <a href="https://commits.webkit.org/299460@main">https://commits.webkit.org/299460@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2394c5535ec827c7c13b8d14c54a624c350af57

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119104 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38785 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29436 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125321 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/71167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a7a63ba0-5ba3-4b30-ae30-65612837f62e) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120982 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39481 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47367 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/90438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/71167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fb3b390f-bd72-4259-ac40-4e2d1e7f32f2) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122057 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/31475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106770 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70892 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/abeb54d9-0a9f-42c7-b930-3b1a731fd168) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/30531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/24884 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68971 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/100921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25067 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128345 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/46011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/34767 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/99044 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46378 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102985 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/98824 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/44288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/22285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/42590 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18959 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45881 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51561 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/45348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/48694 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47033 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
```diff
@@ -524,36 +524,28 @@ static void attemptConnectionInProcessWithoutEntitlement()
-};
-
-static RetainPtr<TestWKWebView> webViewWithOpenInspector(WKWebViewConfiguration *configuration)
+static RetainPtr<TestWKWebView> webViewWithOpenInspector(WKWebViewConfiguration *configuration, id<WKUIDelegate> uiDelegate)
```
It looks like we're not consistent about this in existing code, but it seems like in the majority of cases we insert a space between `id` and `<`.

achristensen07 replied:
There's a WebKit style checker enforcing that we don't add the space:

Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:527: error: [spacing/objc-protocol] Protocol names shouldn't have a space before them.

achristensen07 committed:
Reduce InjectedBundlePageUIClient API surface area
rdar://159335767
https://bugs.webkit.org/show_bug.cgi?id=298012

Reviewed by Jeff Miller.

With rdar://143399667 and rdar://159318803 fixed, most of the callbacks are unused or unreachable.
This removes the unused code, removes tests that were just testing it, and adopts alternatives
for tests that were using it to test other things.

* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::addMessageWithArgumentsToConsole): Deleted.
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration _setShouldSendConsoleLogsToUIProcessForTesting:]):
(-[WKWebViewConfiguration _shouldSendConsoleLogsToUIProcessForTesting]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::addMessageToConsoleForTesting):
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::mouseDidMoveOverElement):
(API::InjectedBundle::PageUIClient::didClickAutoFillButton):
(API::InjectedBundle::PageUIClient::willAddMessageWithArgumentsToConsole): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptAlert): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptConfirm): Deleted.
(API::InjectedBundle::PageUIClient::willRunJavaScriptPrompt): Deleted.
(API::InjectedBundle::PageUIClient::pageDidScroll): Deleted.
(API::InjectedBundle::PageUIClient::didExceedDatabaseQuota): Deleted.
(API::InjectedBundle::PageUIClient::plugInStartLabelTitle const): Deleted.
(API::InjectedBundle::PageUIClient::plugInStartLabelSubtitle const): Deleted.
(API::InjectedBundle::PageUIClient::plugInExtraStyleSheet const): Deleted.
(API::InjectedBundle::PageUIClient::plugInExtraScript const): Deleted.
(API::InjectedBundle::PageUIClient::didResignInputElementStrongPasswordAppearance): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.cpp:
(WebKit::InjectedBundlePageUIClient::willAddMessageToConsole): Deleted.
(WebKit::InjectedBundlePageUIClient::willAddMessageWithArgumentsToConsole): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptAlert): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptConfirm): Deleted.
(WebKit::InjectedBundlePageUIClient::willRunJavaScriptPrompt): Deleted.
(WebKit::InjectedBundlePageUIClient::pageDidScroll): Deleted.
(WebKit::InjectedBundlePageUIClient::didExceedDatabaseQuota): Deleted.
(WebKit::InjectedBundlePageUIClient::didResignInputElementStrongPasswordAppearance): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::addMessageToConsole):
(WebKit::WebChromeClient::runJavaScriptAlert):
(WebKit::WebChromeClient::runJavaScriptConfirm):
(WebKit::WebChromeClient::runJavaScriptPrompt):
(WebKit::WebChromeClient::inputElementDidResignStrongPasswordAppearance):
(WebKit::WebChromeClient::addMessageWithArgumentsToConsole): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::pageDidScroll):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/BundlePageConsoleMessage.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/BundlePageConsoleMessageWithDetails.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ConsoleMessageWithDetails.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DidResignInputElementStrongPasswordAppearance.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::webViewWithOpenInspector):
(TestWebKitAPI::TEST(PrivateClickMeasurement, DaemonDebugMode)):
(TestWebKitAPI::setupSKAdNetworkTest):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetwork)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetworkAboutBlank)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, SKAdNetworkWithoutNavigatingToAppStoreLink)):
(TestWebKitAPI::TEST(PrivateClickMeasurement, NetworkProcessDebugMode)):
(TestWebKitAPI::setInjectedBundleClient): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FTP.mm:
(TestWebKitAPI::TEST(WKWebView, FTPMainResource)):
(TestWebKitAPI::TEST(WKWebView, FTPMainResourceRedirect)):
(TestWebKitAPI::(, )):
(TestWebKitAPI::FTPSubresourceRedirect)):
(TestWebKitAPI::didReceivePageMessageFromInjectedBundle): Deleted.
(TestWebKitAPI::setInjectedBundleClient): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/LegacyPDFPluginTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFTestHelpers.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenTypeDidChange)): Deleted.
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenValueDidChange)): Deleted.
((WebKit, DidResignInputElementStrongPasswordAppearanceWhenFormIsReset)): Deleted.
* Tools/TestWebKitAPI/cocoa/TestUIDelegate.h:
* Tools/TestWebKitAPI/cocoa/TestUIDelegate.mm:
(-[TestUIDelegate _webView:didReceiveConsoleLogForTesting:]):

Canonical link: https://commits.webkit.org/299460@main
